projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/api/meta_job.py

```diff
@@ -10,7 +10,13 @@
 @dataclass
 class SingleJob:
     """
```

Comment:
What's the diff between IStandaloneDataJob and SingleJob?

Reply:
In summary of VDK Core API, 

(1) There is one (primary) **abstraction of what a Data Job is**, and that is `DataJob` class defined run built-in plugin https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/internal/builtin_plugins/run/data_job.py#L233
* It has name, source directory, context, and also could be run.

(2) Then...we have IStandaloneDataJob defined by https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/api/data_job.py#L11

* It has name, source directory, job args, and could be entered.

**The problem is**, those two abstractions obviously describe the same stuff - a Data Job - however, from _OOP perspective_, those interfaces don't have commonalities. That's the gap, that needs to be addressed.

(3) _And, again, this may be a separate design discussion than this PR_,
however it is related, because now we have another MetaJob abstraction, that is actually a Data Job as well. And, again, the same problem of lacking any link to a definitive Data Job interface. What a Data Job is, remains API-incoherent still, and we should address it.

Reply:
Wow, thanks for the detailed information! I totally agree that this should be addressed and that we should find a way to unify the logic instead of dealing with 3 abstractions. I suggest we take this topic into a separate task.